File: LearningPlan.DomainModel/PlanArea.cs
using System.Collections.Generic;

namespace LearningPlan.DomainModel
{
    public class PlanArea : EntityBase
    {
        public string Name { get; set; }
        
        public string PlanId { get; set; }

        public string UserId { get; set; }

        public virtual Plan Plan { get; set; }

        public virtual ICollection<AreaTopic> AreaTopics { get; set; } = new HashSet<AreaTopic>();
    }
}

File: LearningPlan.Web.v2/src/models/plan-area.ts
import type { Topic } from "./topic";

export class PlanArea {
  id: string;
  planId: string;
  name: string;
  areaTopics: Topic[];
  constructor(name: string, planId: string) {
    this.id = '';
    this.planId = planId;
    this.name = name;
    this.areaTopics = [];
  }
}
